File: daq/slc/dqm/src/MonShape.cc
#include "daq/slc/dqm/MonShape.h"

#include <sstream>

using namespace Belle2;

const std::string MonShape::ELEMENT_TAG = "mon-shape";

MonShape::MonShape() :
  MonObject(""),
  m_x(-1), m_y(-1), m_width(-1), m_height(-1) {}

MonShape::MonShape(const std::string& name)
  : MonObject(name),
    m_x(-1), m_y(-1), m_width(-1), m_height(-1) {}

std::string MonShape::toXML() const throw()
{
  std::stringstream ss;
  ss << "<" << ELEMENT_TAG << " "
     << "name='" << m_name << "' ";
  if (m_line_pro != NULL) ss << m_line_pro->toString();
  if (m_fill_pro != NULL) ss << m_fill_pro->toString();
  if (m_font_pro != NULL) ss << m_font_pro->toString();
  if (m_x >= 0 && m_y >= 0 && m_width >= 0 && m_height >= 0) {
    ss << "bounds='" << m_x << "," << m_y << ","
       << m_width << "," << m_height << "' ";
  }
  ss << "/>\n";
  return ss.str();
}

void MonShape::setBounds(float x, float y, float width, float height) throw()
{
  m_x = x;
  m_y = y;
  m_width = width;
  m_height = height;
}

void MonShape::writeConfig(Belle2::Writer& writer) const throw(Belle2::IOException)
{
  MonObject::writeConfig(writer);
}

void MonShape::writeContents(Belle2::Writer& writer) const throw(Belle2::IOException)
{
  writer.writeShort((short)m_fill_color.getRed());
  writer.writeShort((short)m_fill_color.getGreen());
  writer.writeShort((short)m_fill_color.getBlue());
  writer.writeShort((short)m_line_color.getRed());
  writer.writeShort((short)m_line_color.getGreen());
  writer.writeShort((short)m_line_color.getBlue());
}

void MonShape::readConfig(Belle2::Reader& reader) throw(Belle2::IOException)
{
  MonObject::readConfig(reader);
}

void MonShape::readContents(Belle2::Reader& reader) throw(Belle2::IOException)
{
  short red = reader.readShort();
  short green = reader.readShort();
  short blue = reader.readShort();
  m_fill_color.setRGB(red, green, blue);
  red = reader.readShort();
  green = reader.readShort();
  blue = reader.readShort();
  m_line_color.setRGB(red, green, blue);
}

File: daq/slc/extra/backup.javalib/src/b2dqm/core/MonShape.java
package b2dqm.core;

import b2daq.core.Reader;
import b2daq.core.Writer;
import b2daq.graphics.HtmlColor;

public abstract class MonShape extends MonObject {

	private HtmlColor _fill_color = new HtmlColor(HtmlColor.NULL);
	private HtmlColor _line_color = new HtmlColor(HtmlColor.BLACK);

	public MonShape() {
		setName("");
	}

	public MonShape(String name) {
		setName(name);
	}

	public void setFillColor(HtmlColor color) {
		_fill_color = color;
	}

	public void setLineColor(HtmlColor color) {
		_line_color = color;
	}

	public HtmlColor getFillColor() {
		return _fill_color;
	}
	
	public HtmlColor getLineColor() {
		return _line_color;
	}
	
	public void writeConfig(Writer writer) throws Exception {
		super.writeConfig(writer);
	}

	public void writeContents(Writer writer) throws Exception {
		writer.writeShort((short)_fill_color.getRed());
		writer.writeShort((short)_fill_color.getGreen());
		writer.writeShort((short)_fill_color.getBlue());
		writer.writeShort((short)_line_color.getRed());
		writer.writeShort((short)_line_color.getGreen());
		writer.writeShort((short)_line_color.getBlue());
	}

	public void readConfig(Reader reader) throws Exception {
		super.readConfig(reader);
	}

	public void readContents(Reader reader) throws Exception {		
		short red = reader.readShort();
		short green = reader.readShort();
		short blue = reader.readShort();
		_fill_color.setRGB(red, green, blue);
		red = reader.readShort();
		green = reader.readShort();
		blue = reader.readShort();
		_line_color.setRGB(red, green, blue);
	}

}
